### DISCUSSION TITLE
Integrating an External Database

### DISCUSSION THREAD

**Discussion by anonymousyii** [exoplatform/exo-community#30] [Q&A]
Hello fellow community members,

I hope this message finds you well. I am currently exploring the possibility of visualizing data from an external database on the exo community platform.

To give you some background, exo community platform offers a variety of features and customization options. However, I'm unsure about the best approach to integrate data from an external database into our platform's content. This is where I could use your expertise and insights.

Goal: My primary goal is to display specific data from my database in a visually engaging manner. I envision charts, graphs, and tables.

External Database: The data is stored in a postgreSQL database, and I'm wondering how to securely connect to it and fetch the required data

**Comment by rdenarie** [comment]
Hello @anonymousyii 

For his internal database, eXo Community is only supporting mysql database. 

So there are few options for you : 

- If data can be migrated from your external psql database to mysql, you can export then import data in eXo database. After that, you will be able to create a REST service which could read the database an expose data. Then you will be able to create a front end application reading these data. You can follow our [documentation](https://docs.exoplatform.org/guide/developer-guide/intermediate-integration/) to find how to create service an application. In this solution, eXo data and your data are in the same database.
- If data cannot be migrated and must stay in your external database, you can create an application to expose it through REST service. This application will be completly separated from exo. But, you will be able to create a frontend application, in eXo, which will call your REST service to get data, and display it. In this option, eXo data and your data are in separated databases. You could also use tools like [PostgREST](https://postgrest.org/en/stable/) which allow to expose a database as REST. Then you only have the frontend part to develop.
- Another option is to use enterprise version, which allow you to use a psql database. eXo will add his own internal data in your existing database. Then, we are back to the first option, with data in the same database. You will be able to create a REST service which could read the database an expose data. Then you will be able to create a front end application reading these data. 

I hope this answers your questions

Regards,
Romain

**Reply by anonymousyii** [reply]
Thank you @rdenarie, 
 What is the use of the mongodb database in exo?

**Reply by rdenarie** [reply]
MongoDB is used only for the chat module. All chat data are stored into
